### PR TITLE
Ignore Shielded Dust

### PR DESCRIPTION
**Pull request by fluidvanadium**
ignores dust inputs. passes the ignore_dust_inputs test.
prevents the sufficient insufficient fee bug

builds on #1111 and #1113

### CODE REVIEW

File: zingolib/src/wallet/transaction_records_by_id/trait_inputsource.rs [hunk 1]
```diff
@@ -168,6 +175,9 @@ impl InputSource for TransactionRecordsById {
                     Some(missing_value_after_sapling),
                     |rolling_target, (note, note_id)| match rolling_target {
                         Some(targ) if targ == NonNegativeAmount::ZERO => Ok(None),
+                        Some(targ) if note.value().inner() <= MARGINAL_FEE.into_u64() => {
+                            Ok(Some(targ))
```
Comment (zancas) [inline zingolib/src/wallet/transaction_records_by_id/trait_inputsource.rs:185]: What if the spend is cross-pool?   The e.g. 7_000 (> MARGINAL) is still dust if it's transparent to shielded..

Comment (fluidvanadium) [reply zingolib/src/wallet/transaction_records_by_id/trait_inputsource.rs:185]: this is a really good question. it is asking for a good discussion

Comment (fluidvanadium) [reply zingolib/src/wallet/transaction_records_by_id/trait_inputsource.rs:185]: this function is only for shielded pools

**Comment by fluidvanadium** [comment]
superceded by #1142

**Pull request closed without merging**